### ISSUE TITLE
Bug with reorder children products in CategoriesProduct Reorder Controller

### ISSUE THREAD

**Issue by ref** [OFFLINE-GmbH/oc-mall-plugin#765]
On the category edit page in the backend, we can sort the products using the Reorder Controller.

If the category is root, then we see all products, including products in nested categories:

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/controllers/CategoriesProducts.php
<img width="804" alt="Screenshot 2021-05-19 at 18 17 44" src="https://user-images.githubusercontent.com/505388/118804222-9e16cc00-b8ce-11eb-9271-d5821dba31c9.png">

But changing the sorting of a product only works for products that are in an editable category.

Look some screens:

<img width="1437" alt="Screenshot 2021-05-19 at 18 20 43" src="https://user-images.githubusercontent.com/505388/118805017-a15e8780-b8cf-11eb-8fdc-01e9e9c31ad7.png">

<img width="1440" alt="Screenshot 2021-05-19 at 18 23 09" src="https://user-images.githubusercontent.com/505388/118805037-a7ecff00-b8cf-11eb-81c0-803cd4d2d1df.png">

If i change order for any product from reorder page, only products from the root category are saved to the database:
<img width="1197" alt="Screenshot 2021-05-19 at 18 26 29" src="https://user-images.githubusercontent.com/505388/118805121-c0f5b000-b8cf-11eb-8d66-47dc0b8c5cb8.png">

**Comment by ref** [comment]
With debug panel i see what change sorting any child products run db request:

```
update `offline_mall_category_product` set `sort_order` = '0' where `category_id` = '1' and `product_id` = '12'
```

**Comment by georgijk5** [comment]
Hi! Sorry for reopening an issue, but why was this functionality removed on the last version of Mall?

**Comment by tobias-kuendig** [comment]
@georgijk5 I belive it was removed since October 3 contains native controls for reordering relations (you can reorder them in-place by dragging the list items)

**Comment by georgijk5** [comment]
@tobias-kuendig thank you for the swift response!

Unfortunetly that is not the case in Mall 3.5.3.

Will check the 3.5.6 version.

**Comment by georgijk5** [comment]
@tobias-kuendig not available in Mall 3.5.3 either unfortunatelly.

**Comment by tobias-kuendig** [comment]
@georgijk5 this is fixed in v3.5.9, thanks for bringing it to our attention!
